man/fi_pingpong.1.md

```diff
@@ -0,0 +1,148 @@
+
+An endpoint can be specified to be used as the data endpoint. By default, the DGRAM endpoint
+is always selected. Therefore it is designed to transmit data from one node toward the other
+over the selected endpoint, or the default DGRAM if none is selected.
+
+
+# HOW TO RUN TESTS
+
+To launch properly a pingpong test, two termimals have to be open : one for the client,
+the other for the server. However the server must be already running to start the client.
+
+As a client-server test, each have the following usage model:
+
+server$	fi_pingpong -s <server endpoint address>	: start server
+client$	fi_pingpong <server endpoint address>		: connect to server
+
+
+# OPTIONS
+
+The command line of the server and the client must be launched with exactly the same set of options. Otherwise the test would fail because the two nodes would not be able to communicate properly.
+
+## Nodes addressing
+
+*-b <src_port>*
+: The non-default source port number of the endpoint.
+
+*-p <dest_port>*
+: The non-default destination port number of the endpoint.
+
+*-s <src_addr>*
+: The source address.
+
+## Fabric
+
+*-f <provider_name>*
+: The name of the underlying fabric provider e.g. sockets, verbs, psm etc. If the provider name is not provided, the test will pick one from the list of the available providers it finds by fi_getinfo call.
+
+*-e <endpoint> where endpoint = (dgram|rdm|msg)*
+The type of endpoint to be used for data messaging between the two nodes.
+
+*-n <domain>*
+: The name of the specific domain to be used.
+
+## Messaging
+
+*-I <iter>*
+: The number of iterations of the test will run.
+
+*-S <msg_size>*
+: The specific size of the message in bytes the test will use or 'all' to run all the default sizes.
+
+## Utils
+
+*-v*
+: Activate the verification of incoming data.
+
+*-d*
+: Activate output debugging (warning: highly verbose)
+
+*-h*
+: Displays help output for the pingpong test.
+
+
+# USAGE EXAMPLES
+
+## A simple example
+
+	run server: fi_pingpong -f <provider_name> -s <source_addr>
+		server$ fi_pingpong -f sockets -s 192.168.0.123
+	run client: gi_pingpong -f <provider_name> <server_addr>
+		client$	fi_pingpong -f sockets 192.168.0.123
+
+## An example with various options
+
+	run server:
+		server$ fi_pingpong -f usnic -I 1000 -S 1024 -s 192.168.0.123
+	run client:
+		client$ fi_pingpong -f usnic -I 1000 -S 1024 192.168.0.123
+
+
+Specifically, this will run only one test with :
+
+	- usNIC provider
+	- 1000 iterations
+	- 1024 bytes message size
+	- server node as 123.168.0.123
+
+## A longer test
+
+	run server:
+		server$ fi_pingpong -f usnic -I 10000 -S all -s 192.168.0.123
+	run client:
+		client$ fi_pingpong -f usnic -I 10000 -S all 192.168.0.123
+
+## Defaults
+
+For the provider, there is no default value as such. The selection of the provider
+is ultimately determined by libfabric while probing for hardware with the sets of hints generated with the options.
```

Comment:
There is no default provider; if a provider is not specified via the `-f` switch, the test will pick one from the list of available providers (as returned by `fi_getinfo(3)`).